### ISSUE TITLE
Suboptimal GroupNorm Implementation on GPUs

### ISSUE THREAD

**Issue by avik-pal** [avik-pal/Lux.jl#10]
As observed in https://github.com/SciML/FastDEQ.jl/pull/45#issuecomment-1107662055 we get a 2x speedup by moving from GroupNorm to BatchNorm which uses CUDNN kernels.

**Comment by ToucheSir** [comment]
cuDNN should support this, so it's mostly a matter of hooking up NNlib + NNlibCUDA (unless you're fine with directly calling the CUDA.jl routines here)

**Comment by avik-pal** [comment]
Actually I dont think CUDNN supports this (at least could figure it out from its documentation). Pytorch uses its own kernel https://github.com/pytorch/pytorch/blob/35d4a805ebc3b6eca1bafb2d332dffa8d0c1fc54/aten/src/ATen/native/cuda/group_norm_kernel.cu
